wa/output_processors/postgresql.py

```diff
@@ -0,0 +1,484 @@
+        self.current_loid = None
+        self.run_uuid = None
+        self.current_job_uuid = None
+        self.target_uuid = None
+        self.current_artifact_uuid = None
+        self.current_large_object_uuid = None
+        self.current_classifier_uuid = None
+        self.current_metric_uuid = None
+        self.current_augmentation_uuid = None
+        self.current_resource_getter_uuid = None
+        self.current_event_uuid = None
+        self.current_job_aug_uuid = None
+        self.current_parameter_uuid = None
+
+    def initialize(self):
+
+        if not psycopg2:
+            raise ImportError(
+                'The psycopg2 module is required for the ' +
+                'Postgresql Output Processor: {}'.format(import_error_msg))
+        # N.B. Typecasters are for postgres->python and adapters the opposite
+        self.connect_to_database()
+        self.cursor = self.conn.cursor()
+        # Register the adapters and typecasters for enum types
+        self.cursor.execute("SELECT NULL::status_enum")
+        status_oid = self.cursor.description[0][1]
+        self.cursor.execute("SELECT NULL::param_enum")
+        param_oid = self.cursor.description[0][1]
+        LEVEL = psycopg2.extensions.new_type(
+            (status_oid,), "LEVEL", postgres_convert.cast_level)
+        psycopg2.extensions.register_type(LEVEL)
+        PARAM = psycopg2.extensions.new_type(
+            (param_oid,), "PARAM", postgres_convert.cast_vanilla)
+        psycopg2.extensions.register_type(PARAM)
+        psycopg2.extensions.register_adapter(level, postgres_convert.return_as_is(postgres_convert.adapt_level))
+        psycopg2.extensions.register_adapter(
+            postgres_convert.ListOfLevel, postgres_convert.adapt_ListOfX(postgres_convert.adapt_level))
+        psycopg2.extensions.register_adapter(KernelVersion, postgres_convert.adapt_vanilla)
+        psycopg2.extensions.register_adapter(
+            CpuInfo, postgres_convert.adapt_vanilla)
+        psycopg2.extensions.register_adapter(
+            collections.OrderedDict, extras.Json)
+        psycopg2.extensions.register_adapter(dict, extras.Json)
+        psycopg2.extensions.register_adapter(
+            KernelConfig, postgres_convert.create_iterable_adapter(2, explicit_iterate=True))
+        # Register ready-made UUID type adapter
+        extras.register_uuid()
+        # Insert a run_uuid which will be globally accessible during the run
+        self.run_uuid = uuid.UUID(str(uuid.uuid4()))
+        self.cursor.execute(self.sql_command['create_run'], (self.run_uuid,))
+        # Commit cursor commands
+        self.conn.commit()
+
+    def export_job_output(self, output, target_info, run_output):   # pylint: disable=too-many-branches, too-many-statements, too-many-locals
+        self.current_job_uuid = uuid.uuid4()
+        retry_on_status = postgres_convert.ListOfLevel(run_output.run_config.retry_on_status)
+        # Update the run each time
+        self.cursor.execute(
```

Comment:
All of this stuff does not need to be done each time -- info and config stuff doesn't change; Also, those things should be written to the database earlier -- during run initialization, before the first job executes.